src/runtime/knossos.h

```diff
@@ -89,54 +102,54 @@ namespace ks {
 
 	template<typename T0, typename T1, typename T2, typename T3, typename T4, typename ...Ts>
 	struct Tuple<T0, T1, T2, T3, T4, Ts...> {
-		Tuple() { }
-		Tuple(T0 t0, T1 t1, T2 t2, T3 t3, T4 t4, Ts ...ts) : t0(t0), t1(t1), t2(t2), t3(t3), others{t4, ts...} { }
+		KS_INTERFACE Tuple() { }
+		KS_INTERFACE Tuple(T0 t0, T1 t1, T2 t2, T3 t3, T4 t4, Ts ...ts) : t0(t0), t1(t1), t2(t2), t3(t3), others{t4, ts...} { }
 
 		T0 t0; T1 t1; T2 t2; T3 t3;
 		Tuple<T4, Ts...> others;
 	};
 
-	template<typename ...Ts> ks::Tuple<Ts...> make_Tuple(Ts ...ts) {
+	template<typename ...Ts> KS_INTERFACE ks::Tuple<Ts...> make_Tuple(Ts ...ts) {
 		return ks::Tuple<Ts...>{ts...};
 	}
 
 	template<size_t I> struct get_helper_t
 	{
 		template<typename Tuple>
-		static auto const& get(Tuple const& t) {
+		static KS_INTERFACE auto const& get(Tuple const& t) {
 			return get_helper_t<I - 4>::get(t.others);
 		}
 	};
-	template<> struct get_helper_t<0> { template<typename Tuple> static auto const& get(Tuple const& t) { return t.t0; } };
-	template<> struct get_helper_t<1> { template<typename Tuple> static auto const& get(Tuple const& t) { return t.t1; } };
-	template<> struct get_helper_t<2> { template<typename Tuple> static auto const& get(Tuple const& t) { return t.t2; } };
-	template<> struct get_helper_t<3> { template<typename Tuple> static auto const& get(Tuple const& t) { return t.t3; } };
+	template<> struct get_helper_t<0> { template<typename Tuple> static KS_INTERFACE auto const& get(Tuple const& t) { return t.t0; } };
+	template<> struct get_helper_t<1> { template<typename Tuple> static KS_INTERFACE auto const& get(Tuple const& t) { return t.t1; } };
+	template<> struct get_helper_t<2> { template<typename Tuple> static KS_INTERFACE auto const& get(Tuple const& t) { return t.t2; } };
+	template<> struct get_helper_t<3> { template<typename Tuple> static KS_INTERFACE auto const& get(Tuple const& t) { return t.t3; } };
 
-	template<typename T> T& const_cast_deduce_type(T const& t) { return const_cast<T&>(t); }
+	template<typename T> KS_INTERFACE T& const_cast_deduce_type(T const& t) { return const_cast<T&>(t); }
 
-	template<size_t I, typename ...Ts> auto& get(Tuple<Ts...> & t) {
+	template<size_t I, typename ...Ts> KS_INTERFACE auto& get(Tuple<Ts...> & t) {
```

Comment:
I think we could/should have the convention that KS_INTERFACE etc should begin a new line

Reply:
OK done